### DISCUSSION TITLE
Any interest in an R package that interfaces DVC?

### DISCUSSION THREAD

**Discussion by andrewcstewart** [treeverse/dvc#5537] [Show and tell]
Hi all,

I started putting together a little R package that interfaces with the DVC CLI.  You can check it out here: https://github.com/andrewcstewart/dvc-r

It currently only implements a limit set of functionality from DVC, including project setup and data tracking (along with remotes, push, pull, etc).  

One of the primary motives behind this particular package was just that I've been wanting to run through setting up an R package that 1) wraps a python package, 2) wraps a CLI, 3) uses the {testthat} package to unit test that interface, and 4) uses GitHub Actions to build and test the whole thing.  DVC fits that bill well, and I do find myself often using the data tracking features from within R enough that it made a nice little project.

But from here, I'm curious if there is any interest in developing the package further to include other aspects of DVC, etc.  Otherwise I may just leave it as is, but if there are any folks interested enough then it could possibly make for a fun project going forward.

I'd also be curious to hear any thoughts from DVC devs as to the utility of or basic design philosophy of programmatic interfaces to DVC.

Anyway, any thoughts welcome!

-Andrew

**Comment by jendefig** [comment]
Hi @andrewcstewart!  Just seeing this now. As the Community Manger, I see that an intro is in order... 🔗

Another Community member has worked on integrations with R.  Let me introduce you to @jcpsantiago in Github and same in Twitter.  He has created {dvthis} and recently presented it at our [July Office Hours events](https://youtu.be/H22j1lWIvMw).  I know we'd love to see more integrations with R.  Many hands make light work!  Thought there could be a good collaboration between you and Santiago!

**Comment by jcpsantiago** [comment]
@andrewcstewart 👋 hey there! I think our packages are aiming at different goals, but nothing prevents them from being merged into a single effort. I'm def open to it :)

https://github.com/jcpsantiago/dvthis focused more on utilities to create and work in DVC stages, not so much wrap the whole CLI. Let's def chat about this ;)
